src/lib/y2storage/proposal/skip_list_rule.rb

```diff
@@ -0,0 +1,167 @@
+        more_than: [Fixnum].freeze,
+        equal_to:  [Fixnum, Symbol, String].freeze
+      }.freeze
+
+      class << self
+        # Creates a rule from an AutoYaST profile rule definition
+        #
+        # Let's consider the following AutoYaST rule:
+        #
+        #   <listentry>
+        #     <skip_key>size_k</skip_key>
+        #     <skip_value>1048576</skip_value> <!-- translated as string -->
+        #     <skip_if_less_than config:type="boolean">true</skip_if_less_than>
+        #   </listentry>
+        #
+        # @example Building a rule from a hash
+        #   hash # => { "skip_key" => "size_k", "skip_value" => "1048756",
+        #     "skip_if_less_than" => true }
+        #   rule = Proposal::SkipRule.new(hash)
+        #   rule.predicate     #=> :less_than
+        #   rule.raw_reference #=> "1048756"
+        #   rule.key           #=> "size_k"
+        def from_profile_rule(hash)
+          predicate =
+            if hash["skip_if_less_than"]
+              :less_than
+            elsif hash["skip_if_more_than"]
+              :more_than
+            else
+              :equal_to
+            end
+          new(hash["skip_key"], predicate, hash["skip_value"])
+        end
+      end
+
+      # Constructor
+      #
+      # @param key           [String] Name of the attribute to check when applying the rule
+      # @param predicate     [Symbol,String] Comparison (:less_than, :more_than and :equal_to)
+      # @param raw_reference [String] Reference value
+      def initialize(key, predicate, raw_reference)
+        @key = key
+        @predicate = predicate
+        @raw_reference = raw_reference
+      end
+
+      # Determines whether a disk matches the rule
+      #
+      # @param disk [Disk] Disk to match
+      # @return [Boolean] true if the disk matches the rule
+      def matches?(disk)
+        value_from_disk = value(disk)
+        return false unless valid_class?(value_from_disk)
+        send("match_#{predicate}", value_from_disk, cast_reference(raw_reference, value_from_disk.class))
+      end
+
+      # Returns the value to compare from the disk
+      #
+      # This method relies on SkipListValue which is able to gather
+      # the required information from the disk.
+      #
+      # @see Proposal::SkipListValue
+      def value(disk)
+        SkipListValue.new(disk).send(key)
+      end
+
+    private
+
+      # Determines whether the predicate is applicable to the value
+      #
+      # @return [Boolean] true if it is applicable
+      def valid_class?(value)
+        PREDICATES[predicate].include?(value.class)
+      end
+
+      # Cast the reference value in order to do the comparison
+      #
+      # @param raw [String]
+      # @return [String,Fixnum,Symbol] Converted reference value
+      def cast_reference(raw, klass)
+        if klass == Fixnum
+          raw.to_i
+        elsif klass == Symbol
+          raw.respond_to?(:to_sym) ? raw.to_sym : :nothing
```

Comment:
The documentation looks wrong: it could receive also `nil`. I'll updated it.

Regarding the `:nothing` thing, it was just the fallback value used in the [original implementation](https://github.com/yast/yast-autoinstallation/blob/cdfab9e0503bb368d62eba49269c8ae82396d3f0/src/modules/AutoinstStorage.rb#L812).